src/Compilers/CSharp/Test/Symbol/Symbols/SymbolEqualityTests.cs

```diff
@@ -911,6 +911,38 @@ public static void M(A<T> t)
                 );
         }
 
+        [Fact]
+        [WorkItem(58226, "https://github.com/dotnet/roslyn/issues/58226")]
+        public void LambdaSymbol()
+        {
+            var source =
+@"
+#nullable enable
+
+using System;
+using System.Linq;
+
+M1(args => string.Join("" "", args.Select(a => a!.ToString())));
+void M1<TResult>(Func<object?[], TResult>? f) { }
+";
+            var comp = CreateCompilation(source);
+            comp.VerifyDiagnostics();
+
+            var syntaxTree = comp.SyntaxTrees[0];
+            var root = syntaxTree.GetRoot();
+
+            var lambdaSyntax = root.DescendantNodes().OfType<SimpleLambdaExpressionSyntax>().First();
+            var semanticModel1 = comp.GetSemanticModel(syntaxTree);
+            var semanticModel2 = comp.GetSemanticModel(syntaxTree);
+
+            var lambdaSymbol = (IMethodSymbol)semanticModel1.GetSymbolInfo(lambdaSyntax).Symbol;
+            var p1 = lambdaSymbol.Parameters.Single();
+
+            var p2 = semanticModel2.GetDeclaredSymbol(lambdaSyntax.Parameter);
+            VerifyEquality(p1, p2, expectedIncludeNullability: true);
+
+        }
```

Comment:
>} [](http://example.com/codeflow?start=8&length=1)

Let's include a test that specifically tests `Equals()` for distinct lambda expressions with the same span.
```csharp
        [Fact]
        [WorkItem(58226, "[https://github.com/dotnet/roslyn/issues/58226](https://github.com/dotnet/roslyn/issues/58226)")]
        public void LambdaSymbol_02()
        {
            var source =
@"class Program
{
    static void Main()
    {
        var q = from i in new int[] { 4, 5 } where /*pos*/
    }
}";
            var comp = CreateCompilation(source);
            var syntaxTree = comp.SyntaxTrees[0];
            var model = comp.GetSemanticModel(syntaxTree);
            var syntaxNode = syntaxTree.GetRoot().DescendantNodes().OfType<QueryExpressionSyntax>().Single();
            var operation = model.GetOperation(syntaxNode);
            var lambdas = operation.Descendants().OfType<AnonymousFunctionOperation>().Select(op => op.Symbol.GetSymbol<LambdaSymbol>()).ToImmutableArray();
            Assert.Equal(2, lambdas.Length);
            var syntaxRefs = lambdas.SelectAsArray(l => l.SyntaxRef);
            Assert.Equal(syntaxRefs[0].Span, syntaxRefs[1].Span);
            Assert.NotEqual(lambdas[0], lambdas[1]);
        }
```